Management SDK project template and build infrastructure update

allenjzhang opened the pull request:
1. Added management SDK client project template
2. Adjusted Build.Data to support new management SDK projects
3. Added two inline suppression in Azure.Core since it was giving code analysis warnings

allenjzhang committed:
1. Added management SDK client project template, 2. Adjusted Build.Data to support new management SDK projects

pakrym commented on eng/templates/README.md:
```diff
@@ -0,0 +1,16 @@
+# DotNet project templates
+
+## Azure.Management SDK client library template
```
Should this go into the root CONTRIBUTING.md?

pakrym commented on eng/Packages.Data.props:
```diff
@@ -126,6 +126,7 @@
     
     <!-- Azure SDK packages -->
     <PackageReference Update="Azure.Storage.Blobs" Version="12.0.0" />
+
```
nit: revert

pakrym commented on eng/templates/Azure.Management.Template/src/autorest.md:
````diff
@@ -0,0 +1,8 @@
+Run `dotnet msbuild /t:GenerateCode` to generate code.
+
+``` yaml
+input-file:
````
you might want an `azure-arm: true` also added here.

weshaggard commented on eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj:
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Version>0.1.0</Version>
```
Why version 0.1.0? I'd suggest this be `1.0.0-preview.1` as that is what I'd expect folks to start at.

weshaggard commented on eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj:
```diff
@@ -0,0 +1,13 @@
+    <Version>0.1.0</Version>
+    <PackageId>Azure.Management.Template</PackageId>
+    <Description>Azure management client SDK for AzureRP </Description>
+    <PackageTags>azure;management;ProviderNameLowercase</PackageTags>
```
I assume things like ProviderNameLowercase is just a placeholder?

allenjzhang replied:
Yes, it now takes a single parameter say Microsoft.KeyVault or Microsoft.Network. The keyvault, network (lower case) will show up here.

weshaggard commented on eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj:
```diff
@@ -0,0 +1,13 @@
+
+  <PropertyGroup>
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <NoWarn>($NoWarn);CA1812;</NoWarn>
```
If we need to disable this warning for everyone we should probably also move it into the common props.

allenjzhang replied:
CA1812 and AZC0012 specific affect mgmt plane right now. I have moved into common props.

weshaggard commented on eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj:
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
While I don't mind a second template I'd like to understand if there is a good reason for it. Can we converge this with the existing template?

allenjzhang replied:
Are you referring to the SDK\Template folder? Talked to Pavel, the intended purpose is different thus the file content etc. So keeping a separate one. Also that template folder was not set up as dotnet new template.

allenjzhang committed:
1. Addressed comments: moved warning to data.prop, updated initial version, remove extra line
2. added empty tags if none specified
3. added flag to copy ci.yml and test-resource.json into parent folder

allenjzhang committed:
Added link to the dotnet template in CONTRIBUTING.md

pakrym commented on eng/templates/Azure.Management.Template/content/test-resources.json:
```diff
@@ -0,0 +1,4 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0"
```
This is not enough in me experimentation, resources node is required to be present.

allenjzhang replied:
Added.

allenjzhang committed:
Update minimal test-resource ARM template file
